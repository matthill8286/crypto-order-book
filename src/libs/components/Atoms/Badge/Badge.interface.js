
//# sourceMappingURL=Badge.interface.js.map