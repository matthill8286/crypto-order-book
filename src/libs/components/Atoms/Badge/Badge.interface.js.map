{"version":3,"sources":[],"names":[],"mappings":"","sourcesContent":[],"file":"Badge.interface.js"}