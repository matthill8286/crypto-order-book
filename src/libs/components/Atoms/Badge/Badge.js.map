{"version":3,"sources":["../../../../src/components/Atoms/Badge/Badge.tsx"],"names":["React","StyledBadgeLineWrapper","StyledBadgeWrapper","StyledOverflow","StyledTypo","Badge","actionType","badges","isBadgeLine","className","map","badge","i","name"],"mappings":"AAAA,OAAO,KAAKA,KAAZ,MAAuB,OAAvB;AAEA,SACEC,sBADF,EAEEC,kBAFF,EAGEC,cAHF,EAIEC,UAJF;AAOA,OAAO,IAAMC,KAA2B,GAAG,SAA9BA,KAA8B,OAKrC;AAAA,6BAJJC,UAII;AAAA,MAJJA,UAII,gCAJS,UAIT;AAAA,MAHJC,MAGI,QAHJA,MAGI;AAAA,MAFJC,WAEI,QAFJA,WAEI;AAAA,MADJC,SACI,QADJA,SACI;AACJ,SACE,oBAAC,sBAAD;AAAwB,IAAA,WAAW,MAAnC;AAAoC,IAAA,SAAS,EAAEA;AAA/C,KACGF,MAAM,IACLA,MAAM,CAACG,GAAP,CAAW,UAACC,KAAD,EAAQC,CAAR,EAAc;AACvB,QAAI,CAACD,KAAL,EAAY;AACV,aAAO,IAAP;AACD;;AACD,WACE,oBAAC,kBAAD;AACE,MAAA,GAAG,kBAAWA,KAAK,CAACE,IAAjB,cAAyBD,CAAzB,CADL;AAEE,MAAA,UAAU,EAAED,KAAK,CAACL,UAAN,IAAoBA,UAFlC;AAGE,MAAA,OAAO,EAAEM,CAAC,KAAK,CAHjB;AAIE,MAAA,WAAW,EAAEJ;AAJf,OAKE,oBAAC,UAAD;AACE,MAAA,MAAM,EAAC,SADT;AAEE,MAAA,QAAQ,EAAE,KAFZ;AAGE,MAAA,MAAM,EAAC,GAHT;AAIE,MAAA,UAAU,EAAEG,KAAK,CAACL,UAAN,IAAoBA;AAJlC,OAKE,oBAAC,cAAD,QAAiBK,KAAK,CAACE,IAAvB,CALF,CALF,CADF;AAeD,GAnBD,CAFJ,CADF;AAyBD,CA/BM","sourcesContent":["import * as React from 'react'\nimport { BadgeProps } from './Badge.interface'\nimport {\n  StyledBadgeLineWrapper,\n  StyledBadgeWrapper,\n  StyledOverflow,\n  StyledTypo,\n} from './Badge.styled'\n\nexport const Badge: React.FC<BadgeProps> = ({\n  actionType = 'inverted',\n  badges,\n  isBadgeLine,\n  className,\n}) => {\n  return (\n    <StyledBadgeLineWrapper isBadgeLine className={className}>\n      {badges &&\n        badges.map((badge, i) => {\n          if (!badge) {\n            return null\n          }\n          return (\n            <StyledBadgeWrapper\n              key={`badge-${badge.name}-${i}`}\n              actionType={badge.actionType || actionType}\n              isFirst={i === 0}\n              isBadgeLine={isBadgeLine}>\n              <StyledTypo\n                weight=\"regular\"\n                fontSize={'xxs'}\n                margin=\"0\"\n                actionType={badge.actionType || actionType}>\n                <StyledOverflow>{badge.name}</StyledOverflow>\n              </StyledTypo>\n            </StyledBadgeWrapper>\n          )\n        })}\n    </StyledBadgeLineWrapper>\n  )\n}\n"],"file":"Badge.js"}