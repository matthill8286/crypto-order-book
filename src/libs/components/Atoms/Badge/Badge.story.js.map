{"version":3,"sources":["../../../../src/components/Atoms/Badge/Badge.story.tsx"],"names":["number","radios","storiesOf","React","styled","StorybookWrapper","Badge","actionTypes","Prominent","Secondary","Disabled","badges","id","name","actionType","StyledLineMaxWidthContainer","div","knobs","module","add","slice","maxWidth"],"mappings":";;AACA,SAASA,MAAT,EAAiBC,MAAjB,QAA+B,wBAA/B;AACA,SAASC,SAAT,QAA0B,kBAA1B;AACA,OAAO,KAAKC,KAAZ,MAAuB,OAAvB;AACA,SAASC,MAAT;AACA,SAASC,gBAAT;AACA,SAASC,KAAT;AAEA,IAAMC,WAA+C,GAAG;AACtDC,EAAAA,SAAS,EAAE,WAD2C;AAEtDC,EAAAA,SAAS,EAAE,WAF2C;AAGtDC,EAAAA,QAAQ,EAAE;AAH4C,CAAxD;AAMA,IAAMC,MAAM,GAAG,CACb;AACEC,EAAAA,EAAE,EAAE,CADN;AAEEC,EAAAA,IAAI,EAAE;AAFR,CADa,EAKb;AACED,EAAAA,EAAE,EAAE,CADN;AAEEC,EAAAA,IAAI,EAAE;AAFR,CALa,EASb;AACED,EAAAA,EAAE,EAAE,CADN;AAEEC,EAAAA,IAAI,EAAE;AAFR,CATa,EAab;AACED,EAAAA,EAAE,EAAE,CADN;AAEEC,EAAAA,IAAI,EAAE;AAFR,CAba,EAiBb;AACED,EAAAA,EAAE,EAAE,CADN;AAEEC,EAAAA,IAAI,EAAE,UAFR;AAGEC,EAAAA,UAAU,EAAE;AAHd,CAjBa,CAAf;AAwBA,OAAO,IAAMC,2BAA2B,GAAGX,MAAM,CAACY,GAAV;AAAA;AAAA;AAAA,gEAAjC;;AAMP,IAAMC,KAAK,GAAG,SAARA,KAAQ,GAAkB;AAC9B,SAAO;AACLH,IAAAA,UAAU,EAAEb,MAAM,CAAC,MAAD,EAASM,WAAT,EAAsBA,WAAW,CAACC,SAAlC;AADb,GAAP;AAGD,CAJD;;AAMAN,SAAS,CAAC,2BAAD,EAA8BgB,MAA9B,CAAT,CACGC,GADH,CACO,SADP,EACkB,YAAM;AACpB,SACE,oBAAC,gBAAD,QACE,oBAAC,KAAD;AAAO,IAAA,MAAM,EAAER,MAAM,CAACS,KAAP,CAAa,CAAb,EAAgB,CAAhB;AAAf,KAAuCH,KAAK,EAA5C,EADF,CADF;AAKD,CAPH,EAQGE,GARH,CAQO,iBARP,EAQ0B,YAAM;AAC5B,SACE,oBAAC,gBAAD,QACE,oBAAC,KAAD;AAAO,IAAA,MAAM,EAAER,MAAM,CAACS,KAAP,CAAa,CAAb,EAAgB,CAAhB;AAAf,KAAuCH,KAAK,EAA5C,EADF,CADF;AAKD,CAdH,EAeGE,GAfH,CAeO,uBAfP,EAegC,YAAM;AAClC,SACE,oBAAC,gBAAD,QACE,oBAAC,KAAD;AAAO,IAAA,MAAM,EAAER,MAAM,CAACS,KAAP,CAAa,CAAb,EAAgB,CAAhB;AAAf,KAAuCH,KAAK,EAA5C,EADF,CADF;AAKD,CArBH,EAsBGE,GAtBH,CAsBO,WAtBP,EAsBoB,YAAM;AACtB,MAAME,QAAQ,GAAGrB,MAAM,CAAC,4BAAD,EAA+B,GAA/B,CAAvB;;AAEA,MAAMiB,KAAK,GAAG,SAARA,KAAQ,GAAkB;AAC9B,WAAO;AACLH,MAAAA,UAAU,EAAEb,MAAM,CAAC,MAAD,EAASM,WAAT,EAAsBA,WAAW,CAACC,SAAlC;AADb,KAAP;AAGD,GAJD;;AAKA,SACE,oBAAC,gBAAD,QACE,oBAAC,2BAAD;AAA6B,IAAA,KAAK,EAAE;AAAEa,MAAAA,QAAQ,EAARA;AAAF;AAApC,KACE,oBAAC,KAAD;AAAO,IAAA,MAAM,EAAEV,MAAf;AAAuB,IAAA,WAAW;AAAlC,KAAuCM,KAAK,EAA5C,EADF,CADF,CADF;AAOD,CArCH","sourcesContent":["// @ts-nocheck\nimport { number, radios } from '@storybook/addon-knobs'\nimport { storiesOf } from '@storybook/react'\nimport * as React from 'react'\nimport { styled } from '@/styles/styled'\nimport { StorybookWrapper } from '@/utils/StorybookWrapper'\nimport { Badge } from './Badge'\nimport { BadgeActionType, BadgeProps } from './Badge.interface'\nconst actionTypes: { [key: string]: BadgeActionType } = {\n  Prominent: 'prominent',\n  Secondary: 'secondary',\n  Disabled: 'disabled',\n}\n\nconst badges = [\n  {\n    id: 0,\n    name: 'Featured',\n  },\n  {\n    id: 1,\n    name: 'Structured',\n  },\n  {\n    id: 2,\n    name: 'Sponsored',\n  },\n  {\n    id: 3,\n    name: 'Hot Hot Hot',\n  },\n  {\n    id: 4,\n    name: 'Last one',\n    actionType: 'secondary',\n  },\n]\n\nexport const StyledLineMaxWidthContainer = styled.div`\n  background: lightgrey;\n  padding-top: 8px;\n  padding-bottom: 8px;\n`\n\nconst knobs = (): BadgeProps => {\n  return {\n    actionType: radios('Type', actionTypes, actionTypes.Prominent),\n  }\n}\n\nstoriesOf('Design System/Atoms/Badge', module)\n  .add('Default', () => {\n    return (\n      <StorybookWrapper>\n        <Badge badges={badges.slice(0, 1)} {...knobs()} />\n      </StorybookWrapper>\n    )\n  })\n  .add('with two badges', () => {\n    return (\n      <StorybookWrapper>\n        <Badge badges={badges.slice(0, 2)} {...knobs()} />\n      </StorybookWrapper>\n    )\n  })\n  .add('with two action types', () => {\n    return (\n      <StorybookWrapper>\n        <Badge badges={badges.slice(3, 5)} {...knobs()} />\n      </StorybookWrapper>\n    )\n  })\n  .add('BadgeLine', () => {\n    const maxWidth = number('Maximum width of container', 600)\n\n    const knobs = (): BadgeProps => {\n      return {\n        actionType: radios('Type', actionTypes, actionTypes.Prominent),\n      }\n    }\n    return (\n      <StorybookWrapper>\n        <StyledLineMaxWidthContainer style={{ maxWidth }}>\n          <Badge badges={badges} isBadgeLine {...knobs()} />\n        </StyledLineMaxWidthContainer>\n      </StorybookWrapper>\n    )\n  })\n"],"file":"Badge.story.js"}