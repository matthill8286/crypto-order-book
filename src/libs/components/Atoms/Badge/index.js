export * from "./Badge";
export * from "./Badge.interface";
//# sourceMappingURL=index.js.map