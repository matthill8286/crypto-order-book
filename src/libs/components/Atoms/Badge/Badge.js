import * as React from 'react';
import { StyledBadgeLineWrapper, StyledBadgeWrapper, StyledOverflow, StyledTypo } from "./Badge.styled";
export var Badge = function Badge(_ref) {
  var _ref$actionType = _ref.actionType,
      actionType = _ref$actionType === void 0 ? 'inverted' : _ref$actionType,
      badges = _ref.badges,
      isBadgeLine = _ref.isBadgeLine,
      className = _ref.className;
  return React.createElement(StyledBadgeLineWrapper, {
    isBadgeLine: true,
    className: className
  }, badges && badges.map(function (badge, i) {
    if (!badge) {
      return null;
    }

    return React.createElement(StyledBadgeWrapper, {
      key: "badge-".concat(badge.name, "-").concat(i),
      actionType: badge.actionType || actionType,
      isFirst: i === 0,
      isBadgeLine: isBadgeLine
    }, React.createElement(StyledTypo, {
      weight: "regular",
      fontSize: 'xxs',
      margin: "0",
      actionType: badge.actionType || actionType
    }, React.createElement(StyledOverflow, null, badge.name)));
  }));
};
//# sourceMappingURL=Badge.js.map