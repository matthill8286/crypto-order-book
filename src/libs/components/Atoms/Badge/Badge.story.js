function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { number, radios } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { styled } from "../../../styles/styled";
import { StorybookWrapper } from "../../../utils/StorybookWrapper";
import { Badge } from "./Badge";
var actionTypes = {
  Prominent: 'prominent',
  Secondary: 'secondary',
  Disabled: 'disabled'
};
var badges = [{
  id: 0,
  name: 'Featured'
}, {
  id: 1,
  name: 'Structured'
}, {
  id: 2,
  name: 'Sponsored'
}, {
  id: 3,
  name: 'Hot Hot Hot'
}, {
  id: 4,
  name: 'Last one',
  actionType: 'secondary'
}];
export var StyledLineMaxWidthContainer = styled.div.withConfig({
  displayName: "Badgestory__StyledLineMaxWidthContainer",
  componentId: "sc-1o7otni-0"
})(["background:lightgrey;padding-top:8px;padding-bottom:8px;"]);

var knobs = function knobs() {
  return {
    actionType: radios('Type', actionTypes, actionTypes.Prominent)
  };
};

storiesOf('Design System/Atoms/Badge', module).add('Default', function () {
  return React.createElement(StorybookWrapper, null, React.createElement(Badge, _extends({
    badges: badges.slice(0, 1)
  }, knobs())));
}).add('with two badges', function () {
  return React.createElement(StorybookWrapper, null, React.createElement(Badge, _extends({
    badges: badges.slice(0, 2)
  }, knobs())));
}).add('with two action types', function () {
  return React.createElement(StorybookWrapper, null, React.createElement(Badge, _extends({
    badges: badges.slice(3, 5)
  }, knobs())));
}).add('BadgeLine', function () {
  var maxWidth = number('Maximum width of container', 600);

  var knobs = function knobs() {
    return {
      actionType: radios('Type', actionTypes, actionTypes.Prominent)
    };
  };

  return React.createElement(StorybookWrapper, null, React.createElement(StyledLineMaxWidthContainer, {
    style: {
      maxWidth: maxWidth
    }
  }, React.createElement(Badge, _extends({
    badges: badges,
    isBadgeLine: true
  }, knobs()))));
});
//# sourceMappingURL=Badge.story.js.map