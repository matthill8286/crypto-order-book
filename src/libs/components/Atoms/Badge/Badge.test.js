import React from 'react';
import { Badge } from "./Badge";
import { renderWithTheme } from "../../../testRenderer";
var mockedBadges = [{
  id: 0,
  name: 'Badge No.1'
}, {
  id: 1,
  name: 'Some other label'
}, {
  id: 2,
  name: 'Free shipping'
}, {
  id: 3,
  name: 'Hot Hot Hot'
}, {
  id: 4,
  name: 'Last one'
}];
describe('Badge', function () {
  it('should render', function () {
    var tree = renderWithTheme(React.createElement(Badge, {
      badges: mockedBadges
    }));
    expect(tree).toMatchSnapshot();
  });
  it('should render with primary color', function () {
    var tree = renderWithTheme(React.createElement(Badge, {
      badges: mockedBadges,
      actionType: "prominent"
    }));
    expect(tree).toMatchSnapshot();
  });
  it('should render with secondary color', function () {
    var tree = renderWithTheme(React.createElement(Badge, {
      badges: mockedBadges,
      actionType: "secondary"
    }));
    expect(tree).toMatchSnapshot();
  });
  it('should render a badge line', function () {
    var tree = renderWithTheme(React.createElement(Badge, {
      isBadgeLine: true,
      badges: mockedBadges
    }));
    expect(tree).toMatchSnapshot();
  });
  it('should render a badge line secondary', function () {
    var tree = renderWithTheme(React.createElement(Badge, {
      badges: mockedBadges,
      isBadgeLine: true,
      actionType: "secondary"
    }));
    expect(tree).toMatchSnapshot();
  });
  it('should render a badge line disabled', function () {
    var tree = renderWithTheme(React.createElement(Badge, {
      badges: mockedBadges,
      isBadgeLine: true,
      actionType: "disabled"
    }));
    expect(tree).toMatchSnapshot();
  });
});
//# sourceMappingURL=Badge.test.js.map