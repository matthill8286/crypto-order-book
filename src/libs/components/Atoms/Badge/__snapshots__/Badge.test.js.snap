// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render 1`] = `
<div
  className="Badgestyled__StyledBadgeLineWrapper-sc-1xo0r7x-1 ejODHn"
>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 gQWEsv"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Badge No.1
      </div>
    </span>
  </div>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 gQWEsv"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Some other label
      </div>
    </span>
  </div>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 gQWEsv"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Free shipping
      </div>
    </span>
  </div>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 gQWEsv"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Hot Hot Hot
      </div>
    </span>
  </div>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 gQWEsv"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Last one
      </div>
    </span>
  </div>
</div>
`;

exports[`Badge should render a badge line 1`] = `
<div
  className="Badgestyled__StyledBadgeLineWrapper-sc-1xo0r7x-1 ejODHn"
>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 gQWEsv"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Badge No.1
      </div>
    </span>
  </div>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 gQWEsv"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Some other label
      </div>
    </span>
  </div>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 gQWEsv"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Free shipping
      </div>
    </span>
  </div>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 gQWEsv"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Hot Hot Hot
      </div>
    </span>
  </div>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 gQWEsv"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Last one
      </div>
    </span>
  </div>
</div>
`;

exports[`Badge should render a badge line disabled 1`] = `
<div
  className="Badgestyled__StyledBadgeLineWrapper-sc-1xo0r7x-1 ejODHn"
>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 ecZGlg"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Badge No.1
      </div>
    </span>
  </div>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 ecZGlg"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Some other label
      </div>
    </span>
  </div>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 ecZGlg"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Free shipping
      </div>
    </span>
  </div>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 ecZGlg"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Hot Hot Hot
      </div>
    </span>
  </div>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 ecZGlg"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Last one
      </div>
    </span>
  </div>
</div>
`;

exports[`Badge should render a badge line secondary 1`] = `
<div
  className="Badgestyled__StyledBadgeLineWrapper-sc-1xo0r7x-1 ejODHn"
>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 klOziU"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Badge No.1
      </div>
    </span>
  </div>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 klOziU"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Some other label
      </div>
    </span>
  </div>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 klOziU"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Free shipping
      </div>
    </span>
  </div>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 klOziU"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Hot Hot Hot
      </div>
    </span>
  </div>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 klOziU"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Last one
      </div>
    </span>
  </div>
</div>
`;

exports[`Badge should render with primary color 1`] = `
<div
  className="Badgestyled__StyledBadgeLineWrapper-sc-1xo0r7x-1 ejODHn"
>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 ePrcMg"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Badge No.1
      </div>
    </span>
  </div>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 ePrcMg"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Some other label
      </div>
    </span>
  </div>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 ePrcMg"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Free shipping
      </div>
    </span>
  </div>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 ePrcMg"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Hot Hot Hot
      </div>
    </span>
  </div>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 ePrcMg"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Last one
      </div>
    </span>
  </div>
</div>
`;

exports[`Badge should render with secondary color 1`] = `
<div
  className="Badgestyled__StyledBadgeLineWrapper-sc-1xo0r7x-1 ejODHn"
>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 klOziU"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Badge No.1
      </div>
    </span>
  </div>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 klOziU"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Some other label
      </div>
    </span>
  </div>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 klOziU"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Free shipping
      </div>
    </span>
  </div>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 klOziU"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Hot Hot Hot
      </div>
    </span>
  </div>
  <div
    className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 klOziU"
      fontSize="xxs"
    >
      <div
        className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
      >
        Last one
      </div>
    </span>
  </div>
</div>
`;
